src/main/java/mt/edu/um/cf2/jgribx/Bytes2Number.java

```diff
@@ -118,112 +121,100 @@ public static float bytesToFloat(byte[] bytes, int format)
-   }
-
-
-   /**
-    * Convert two bytes into an unsigned integer.
-    *
-    * @param a higher byte
-    * @param b lower byte
-    *
-    * @return integer value
-    */
-   public static int uint2(int a, int b)
-   {
-
-      return a << 8 | b;
-   }
-
-
-   /**
-    * Convert three bytes into an unsigned integer.
-    *
-    * @param a higher byte
-    * @param b middle byte
-    * @param c lower byte
-    *
-    * @return integer value
-    */
-   public static int uint3(int a, int b, int c)
-   {
-
-      return a << 16 | b << 8 | c;
-   }
-
-
-   /**
-    * Convert four bytes into a float value.
-    *
-    * @param a highest byte
-    * @param b higher byte
-    * @param c lower byte
-    * @param d lowest byte
-    *
-    * @return float value
-    */
-   public static float float4(int a, int b, int c, int d)
-   {
+
+    /**
+     * Convert two bytes into a signed integer.
+     *
+     * @param a higher byte
+     * @param b lower byte
+     * @return integer value
+     */
+    public static int int2(int a, int b)
+    {
+
+        return (1 - ((a & 128) >> 6)) * ((a & 127) << 8 | b);
+    }
+
+    /**
+     * Convert three bytes into a signed integer.
+     *
+     * @param a higher byte
+     * @param b middle part byte
+     * @param c lower byte
+     * @return integer value
+     */
+    public static int int3(int a, int b, int c)
+    {
+
+        return (1 - ((a & 128) >> 6)) * ((a & 127) << 16 | b << 8 | c);
+    }
+
+    /**
+     * Convert four bytes into a signed integer.
+     *
+     * @param a highest byte
+     * @param b higher middle byte
+     * @param c lower middle byte
+     * @param d lowest byte
+     * @return integer value
+     */
+    public static int int4(int a, int b, int c, int d)
+    {
+
+        return (1 - ((a & 128) >> 6)) * ((a & 127) << 24 | b << 16 | c << 8 | d);
+    }
+
+    /**
+     * Convert two bytes into an unsigned integer.
+     *
+     * @param a higher byte
+     * @param b lower byte
+     * @return integer value
+     */
+    public static int uint2(int a, int b)
+    {
+
+        return a << 8 | b;
+    }
+
+    /**
+     * Convert three bytes into an unsigned integer.
+     *
+     * @param a higher byte
+     * @param b middle byte
+     * @param c lower byte
+     * @return integer value
+     */
+    public static int uint3(int a, int b, int c)
+    {
+
+        return a << 16 | b << 8 | c;
+    }
+
+    /**
+     * Convert four bytes into a float value.
+     *
+     * @param a highest byte
+     * @param b higher byte
+     * @param c lower byte
+     * @param d lowest byte
+     * @return float value
+     */
+    public static float float4(int a, int b, int c, int d)
+    {
        /*
         byte test[] = {(byte)a, (byte) b, (byte) c, (byte) d};
         return ByteBuffer.wrap(test).getFloat();
-        */  
-      int sgn, mant, exp;
-
-      mant = b << 16 | c << 8 | d;
-      if (mant == 0) return 0.0f;
+        */
```

Comment:
I'd be in favor of removing this commented-out code (IMO there's rarely a good reason to leave commented-out code lying around when we have version control for that).

Reply:
Did a quick pass over all the files and removed commented out code and did general cleanup